src/libImaging/Convert.c

```diff
@@ -310,40 +310,46 @@ rgb2bgr24(UINT8 *out, const UINT8 *in, int xsize) {
 }
 
 static void
-rgb2hsv_row(UINT8 *out, const UINT8 *in) {  // following colorsys.py
-    float h, s, rc, gc, bc, cr;
-    UINT8 maxc, minc;
-    UINT8 r, g, b;
-    UINT8 uh, us, uv;
-
-    r = in[0];
-    g = in[1];
-    b = in[2];
-    maxc = MAX(r, MAX(g, b));
-    minc = MIN(r, MIN(g, b));
-    uv = maxc;
+rgb2hsv_row(UINT8 *out, const UINT8 *in) {
+    // based on Python's colorsys module
+
+    const UINT8 r = in[0];
+    const UINT8 g = in[1];
+    const UINT8 b = in[2];
+
+    const UINT8 maxc = MAX(r, MAX(g, b));
+    const UINT8 minc = MIN(r, MIN(g, b));
+
+    UINT8 uh, us;
+    const UINT8 uv = maxc;
+
     if (minc == maxc) {
         uh = 0;
         us = 0;
     } else {
-        cr = (float)(maxc - minc);
-        s = cr / (float)maxc;
-        rc = ((float)(maxc - r)) / cr;
-        gc = ((float)(maxc - g)) / cr;
-        bc = ((float)(maxc - b)) / cr;
+        const UINT8 color_range = maxc - minc;
+        double h;
+
+        const double cr = (double)color_range;
```

Comment:
I believe double in overkill here.

```suggestion
        float h;

        const float cr = (float)color_range;
```

Reply:
Is there any downside to using a `double` here?

Reply:
Performance, of course. It could be more or less noticeable depending on platform. Also long types often prevents auto vectorisation (which I believe still doesn't work here even with floats).

```python
from PIL import Image
im = Image.new('RGB', (1024, 1024))
%timeit -n 1000 im.convert('HSV')
```

### Master
2.04 ms ± 39.5 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

### Yay295:patch-2
2.18 ms ± 27.2 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

### Yay295:patch-2 with float
2.05 ms ± 62.6 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)